dragonfly/entity/flammable.go

```diff
@@ -0,0 +1,17 @@
+package entity
+
+// Flammable is an interface for entities that can be set on fire.
+type Flammable interface {
+	// FireProof is whether the entity is currently fire proof.
+	FireProof() bool
+	// FireDamage deals fire damage to the entity.
+	FireDamage(amount float64)
+	// LavaDamage deals lava damage to the entity.
+	LavaDamage(amount float64)
```

Comment:
Can we somehow get rid of this method and the method above? If we can move this to an internal package to get around import cycles that would be great. The methods by themselves are pretty useless.